clang/runtime/dpct-rt/include/dpct/switcher.hpp

```diff
@@ -0,0 +1,111 @@
+//==---- switcher.hpp -----------------------------*- C++ -*----------------==//
+//
+// Copyright (C) Intel Corporation
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __DPCT_SWITCHER_HPP__
+#define __DPCT_SWITCHER_HPP__
+
+// Only for testing, it need be removed after update dnnl's helper function test
+#include <dpct/dpct.hpp>
+
+namespace dpct {
+namespace detail {
+namespace switcher {
+#if USE_DPCT_HELPER
+namespace chosen_ns = ::dpct;
+template <typename T> using DataType = ::dpct::DataType<T>;
+using memcpy_direction = ::dpct::memcpy_direction;
+template <class... Args> using kernel_name = dpct_kernel_name<Args...>;
+#else
+namespace chosen_ns = ::syclcompat;
+template <typename T> using DataType = ::syclcompat::detail::DataType<T>;
+using memcpy_direction = ::syclcompat::experimental::memcpy_direction;
+template <class... Args> using kernel_name = syclcompat_kernel_name<Args...>;
+#endif
+
+using chosen_ns::get_current_device;
+using chosen_ns::get_default_context;
+using chosen_ns::queue_ptr;
+using chosen_ns::detail::get_pointer_attribute;
+using chosen_ns::detail::pointer_access_attribute;
```

Comment:
please make sure the backward compatiblity  (migrated code can be built using latest helper function).

Reply:
Using the new namespace `dpct::cs` will not break the backward compatibility since the using declaration added those `dpct` namespace types into the `dpct::cs` namespace.